odbc/src/api/functions.rs

```diff
@@ -2211,12 +2204,92 @@ unsafe fn sql_get_infow_helper(
                 // Setting this to 10, which is our default for the number of workers in the mongo driver's connection pool.
                 InfoType::SQL_MAX_CONCURRENT_ACTIVITIES => {
                     i16_len::set_output_fixed_data(&10, info_value_ptr, string_length_ptr)
                 }
-                InfoType::SQL_DTC_TRANSITION_COST => {
-                    i16_len::set_output_fixed_data(&0, info_value_ptr, string_length_ptr)
+                InfoType::SQL_FORWARD_ONLY_CURSOR_ATTRIBUTES1
+                | InfoType::SQL_KEYSET_CURSOR_ATTRIBUTES1
+                | InfoType::SQL_DYNAMIC_CURSOR_ATTRIBUTES1
+                | InfoType::SQL_STATIC_CURSOR_ATTRIBUTES1 => {
+                    i16_len::set_output_fixed_data(
+                        &SQL_CA1_NEXT,
+                        info_value_ptr,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_FORWARD_ONLY_CURSOR_ATTRIBUTES2
+                | InfoType::SQL_KEYSET_CURSOR_ATTRIBUTES2
+                | InfoType::SQL_DYNAMIC_CURSOR_ATTRIBUTES2
+                | InfoType::SQL_STATIC_CURSOR_ATTRIBUTES2 => {
+                    i16_len::set_output_fixed_data(
+                        &MONGO_CA2_SUPPORT,
+                        info_value_ptr,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_SCROLL_OPTIONS => {
+                    i16_len::set_output_fixed_data(
+                        &MONGO_SO_SUPPORT,
+                        info_value_ptr,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_NEED_LONG_DATA_LEN => {
+                    i16_len::set_output_wstring_as_bytes(
+                        SQL_INFO_Y,
+                        info_value_ptr,
+                        buffer_length as usize,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_TXN_ISOLATION_OPTION => {
+                    i16_len::set_output_fixed_data(
+                        &SQL_TXN_SERIALIZABLE,
+                        info_value_ptr,
+                        string_length_ptr,
+                    )
+                }
+                InfoType::SQL_DATABASE_NAME => {
+                    let conn = must_be_valid!((*conn_handle).as_connection());
+                    let attributes = conn.attributes.read().unwrap();
+                    if attributes.current_catalog.is_some() {
+                        i16_len::set_output_wstring_as_bytes(
+                            attributes.current_catalog.as_ref().unwrap().as_str(),
+                            info_value_ptr,
+                            buffer_length as usize,
+                            string_length_ptr,
+                        )
+                    } else {
+                        err = Some(ODBCError::ConnectionNotOpen);
+                        SqlReturn::ERROR
```

Comment:
Just a note here
I checked and this should be pre-covered by the DM.

08003 | Connection not open | (DM) The type of information requested in InfoType requires  an open connection. Of the information types reserved by ODBC, only  SQL_ODBC_VER can be returned without an open connection.

Reply:
Sounds good, I like keeping this since rust forces us to have an else block, anyway. I guess I could add unreachable, but who knows if one of the other DMs is wrong (looking at you iodbc :)).